src/tests/e2e/mocks/transfers.js

```diff
@@ -0,0 +1,207 @@
+  // TODO: add btc tx hash
+}
+
+type TransferMockSpecType = {
+  cryptoType: string,
+
+  // ACCOUNTS constant for details
+  senderAccountType: string,
+  receiverAccountType?: string,
+
+  // a state from transferStates
+  state: string
+}
+
+function generateData (
+  transferOut: Array<TransferMockSpecType> = [],
+  transferIn: Array<TransferMockSpecType> = []
+) {
+  let transferDataList = []
+  let driveTransferHistory = {}
+  for (let [idx, transfer] of transferOut.entries()) {
+    // deep copy
+    let transferData = JSON.parse(JSON.stringify(TRANSFER_DATA_BASE))
+
+    // set ids
+    transferData.transferId = TRANSFER_ID_LIST[idx]
+    transferData.receivingId = RECEIVING_ID_LIST[idx]
+
+    // set basic transfer info
+    transferData.cryptoType = transfer.cryptoType
+    transferData.senderAccount = ACCOUNTS[transfer.senderAccountType]
+    if (transfer.receiverAccountType) {
+      transferData.receiverAccount = ACCOUNTS[transfer.receiverAccountType]
+    }
+
+    // set state-dependent vars
+    switch (transfer.state) {
+      case 'SEND_PENDING':
+        transferData.sendTxHash = TX_HASH[transfer.cryptoType].txHash
+        transferData.transferStage = 'SenderToChainsfer'
+        transferData.senderToChainsfer = {
+          ...TX_HASH[transfer.cryptoType],
+          txState: 'Pending'
+        }
+        transferData.inEscrow = 0
+        transferData.expired = false
+        break
+      case 'SEND_CONFIRMED_RECEIVE_PENDING':
+        transferData.sendTxHash = TX_HASH[transfer.cryptoType].txHash
+        transferData.transferStage = 'SenderToChainsfer'
+        transferData.senderToChainsfer = {
+          ...TX_HASH[transfer.cryptoType],
+          txState: 'Confirmed'
+        }
+        transferData.inEscrow = 1
+        transferData.expired = false
+        break
+      case 'SEND_CONFIRMED_EXPIRED_RECEIVE_CONFIRMED':
+        transferData.sendTxHash = TX_HASH[transfer.cryptoType].txHash
+        transferData.transferStage = 'chainsfrToReceiver'
+        transferData.senderToChainsfer = {
+          ...TX_HASH[transfer.cryptoType],
+          txState: 'Confirmed'
+        }
+        transferData.chainsferToReceiver = {
+          ...TX_HASH[transfer.cryptoType],
+          txState: 'Confirmed'
+        }
+        transferData.inEscrow = 0
+        transferData.expired = false
+        break
+      // TODO: support more states
+      default:
+        throw new Error(`Transfer state ${transfer.state} not supported`)
+    }
+
+    transferDataList.push(transferData)
+    driveTransferHistory[transferData.transferId] = {
+      ...ENCRYPTED_PRIVATE_KEYS[transferData.cryptoType],
+      transferId: transferData.transferId,
+      senderAccountId: transferData.senderAccount,
+      sendTimestamp: TX_HASH[transferData.cryptoType].txTimestamp,
+      sendTxHash: TX_HASH[transferData.cryptoType].txHash
+    }
+  }
+  // TODO: support transferIn
+
+  return { transferDataList, driveTransferHistory }
+}
+
+const DEFAULT_DATA = {
```

Comment:
DEFALT_TRANSFER_DATA. Name it DEFAULT_DATA can cause confusion when it's imported by other files

Reply:
fixed